subprojects/dependency-management/src/main/java/org/gradle/internal/component/external/model/DefaultMavenModuleResolveMetadata.java

```diff
@@ -92,4 +103,42 @@ public String getSnapshotTimestamp() {
         return variants;
     }
 
+    @Nullable
+    @Override
+    public AttributesSchemaInternal getAttributesSchema() {
+        return new PreferJavaRuntimeVariant();
```

Comment:
:o: If it's stateless, let's make it static: dependency resolution already creates way too many objects.

Reply:
My intent was to defer the creation of this until it was needed. But this wasn't fully thought through.

I'll make the instance static, and assume of creating on load is negligible.